src/main/java/com/metamx/http/client/HttpClientConfig.java

```diff
@@ -24,6 +24,35 @@
  */
 public class HttpClientConfig
 {
+  public enum CompressionCodec
+  {
+    IDENTITY {
+      @Override
+      public String getEncodingString()
+      {
+        return "identity";
+      }
+    },
+    GZIP {
+      @Override
+      public String getEncodingString()
+      {
+        return "gzip";
+      }
+    },
+    DEFLATE {
+      @Override
+      public String getEncodingString()
+      {
+        return "deflate";
+      }
+    };
+
+    public abstract String getEncodingString();
```

Comment:
Can remove this method and use `compressionCodec.name().toLowerCase()` instead of `getEncodingString()`. Or even call enum constants lowercase and use just `name()`.